### PR TITLE
Samplefix maccopypaste

### PR DESCRIPTION
**Pull request by amitgosavi1**
[QAXM] fix bug https://bugzilla.xamarin.com/show_bug.cgi?id=45890

Fixed crash on copy button.
Steps to reproduce:
1. Launch app, select image.
2. Click copy button, this will update pasteboard.
3. Close application & relaunch it.
4. Click on paste button. This will paste earlier copied image.
5. Click on copy button. App was crashing here.

### CODE REVIEW

File: MacCopyPaste/MacCopyPaste/Classes/ImageDocument.cs [hunk 1]
```diff
@@ -90,12 +90,15 @@ public ImageDocument ()
 			if (ok) {
 				// Read the image off of the pasteboard
 				NSObject [] objectsToPaste = pasteboard.ReadObjectsForClasses (classArray2, null);
-				ImageInfo info = (ImageInfo)objectsToPaste[0];
+                	if (objectsToPaste.Length > 0)
```
Comment (chamons) [inline MacCopyPaste/MacCopyPaste/Classes/ImageDocument.cs]: Tab indent is wrong here.

Comment (chamons) [reply MacCopyPaste/MacCopyPaste/Classes/ImageDocument.cs]: Also, info doesn't appear be used?

File: MacCopyPaste/MacCopyPaste/Classes/ImageInfo.cs [hunk 2]
```diff
@@ -34,18 +34,23 @@ public ImageInfo(NSCoder decoder)
 			var type = decoder.DecodeObject("imageType") as NSString;
 
 			// Save data
-			Name = name.ToString();
-			ImageType = type.ToString ();
+            		if (name!= null && type != null)
```
Comment (chamons) [inline MacCopyPaste/MacCopyPaste/Classes/ImageInfo.cs]: space before !=

File: MacCopyPaste/MacCopyPaste/Classes/ImageInfo.cs [hunk 3]
```diff
@@ -34,18 +34,23 @@ public ImageInfo(NSCoder decoder)
...
 		#region Public Methods
 		[Export ("encodeWithCoder:")]
 		public void EncodeTo (NSCoder encoder)
 		{
-			// Encode data
-			encoder.Encode(new NSString(Name),"name");
-			encoder.Encode(new NSString(ImageType),"imageType");
+            		if (Name != null && ImageType != null)
+            		{
+                		encoder.Encode(new NSString(Name), "name");
```
Comment (chamons) [inline MacCopyPaste/MacCopyPaste/Classes/ImageInfo.cs]: space before (

Please check your formatting settings.

Comment (amitgosavi1) [reply MacCopyPaste/MacCopyPaste/Classes/ImageInfo.cs]: Formatting updated.

File: MacCopyPaste/MacCopyPaste/Classes/ImageInfo.cs (hunk 2 above)
Comment (amitgosavi1) [reply MacCopyPaste/MacCopyPaste/Classes/ImageInfo.cs]: Formatting updated.

File: MacCopyPaste/MacCopyPaste/Classes/ImageDocument.cs (hunk 1 above)
Comment (amitgosavi1) [reply MacCopyPaste/MacCopyPaste/Classes/ImageDocument.cs]: Formatting updated. info is not used anywhere but objectsToPaste is type-casted which was crashing earlier.

File: MacCopyPaste/MacCopyPaste/Classes/ImageDocument.cs [hunk 4]
```diff
@@ -89,13 +89,13 @@ public ImageDocument ()
...
 			if (ok) {
 				// Read the image off of the pasteboard
-				NSObject [] objectsToPaste = pasteboard.ReadObjectsForClasses (classArray2, null);
-				ImageInfo info = (ImageInfo)objectsToPaste[0];
-
-
+				NSObject[] objectsToPaste = pasteboard.ReadObjectsForClasses(classArray2, null);
+				if (objectsToPaste.Length > 0)
+				{
+					ImageInfo info = (ImageInfo)objectsToPaste[0];
```
Comment (chamons) [inline MacCopyPaste/MacCopyPaste/Classes/ImageDocument.cs]: This still feels incomplete. What is it doing here?

Comment (amitgosavi1) [reply MacCopyPaste/MacCopyPaste/Classes/ImageDocument.cs]: This is as per the sample submitted. - It just reading image from the pasteboard & get the imageInfo.

Comment (chamons) [reply MacCopyPaste/MacCopyPaste/Classes/ImageDocument.cs]: ¯\_(ツ)_/¯ 

Good enough I guess then.

**Pull request merged**